### PR TITLE
Fix #8840: Tenant domain name is not accepted by Key Vault

### LINKED ISSUE #8840: az keyvault create gives "Bad JSON content found in the request."
I'm trying to create a new keyvault using AZ CLI. My command line is as follows:

```powershell
 az keyvault create -n 'comet-key-vault' -g 'jun-mgmt' --enable-soft-delete true --enabled-for-deployment true --enabled-for-template-deployment true 
```
the response I get is:
```powershell
Bad JSON content found in the request.
```

I am using AZ CLI on Windows, versions as follows:

```powershell
azure-cli                         2.0.60
keyvault                          2.2.12
```

I am not able to see what is wrong in the request. Anyone here who can help?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 226e5769-f6c8-a480-db78-727fb96348ec
* Version Independent ID: 8ee06a2a-6eda-11be-3c8e-6ad948be207e
* Content: [az keyvault](https://docs.microsoft.com/en-us/cli/azure/keyvault?view=azure-cli-latest)
* Content Source: [src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_help.py](https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_help.py)
* Service: **key-vault**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

### PR DESCRIPTION
**Pull request by jiasli**
Fix #8840: Convert tenant domain name to GUID if it is not.

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/keyvault/custom.py [hunk 1]
```diff
@@ -28,6 +29,28 @@
...
+
+
+# Convert tenant domain name to GUID
+# https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-protocols-oidc#fetch-the-openid-connect-metadata-document
+def _to_guid_tenant_id(tenant_id):
+
+    if not _is_guid(tenant_id):
+        import requests
+        url = 'https://login.microsoftonline.com/{}/.well-known/openid-configuration'.format(tenant_id)
+        metadata = requests.get(url).json()
```
Comment (yugangw-msft) [inline src/azure-cli/azure/cli/command_modules/keyvault/custom.py]: Please handle the case when a proxy is involved, check out the [code](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/util.py#L573).
Also like I mentioned endpoint like `login.microsoftonline.com` should come from current cloud configuration, again code is [here](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/util.py#L556).
Last, can we do the conversion during `az login`?

Comment (rajarshi-singh) [reply src/azure-cli/azure/cli/command_modules/keyvault/custom.py]: +1 for doing the conversion at 'az login' level if possible.

Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/keyvault/custom.py]: I have moved the domain resolution logic to `az login`. @yugangw-msft, please kindly review. Thanks!

File: src/azure-cli/azure/cli/command_modules/profile/_validators.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+logger = get_logger(__name__)
+
+
+def _is_guid(guid):
+    try:
+        uuid.UUID(guid)
+        return True
+    except ValueError:
+        pass
```
Comment (yugangw-msft) [inline src/azure-cli/azure/cli/command_modules/profile/_validators.py]: you can `return False` here

**Comment by yugangw-msft** [comment]
This is a user facing change, please add a releasing note

File: src/azure-cli/azure/cli/command_modules/profile/_validators.py (hunk 2 above)
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/profile/_validators.py]: Good point! This is actually copied from https://github.com/Azure/azure-cli/blob/fa3258bb1de15383e4c9568f039ff2440972c5c3/src/azure-cli/azure/cli/command_modules/role/custom.py#L1659 and other places. I will fix this by now.

File: src/azure-cli/azure/cli/command_modules/profile/_validators.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+def validate_tenant(cmd, namespace):
+    """
+    Make sure tenant is a GUID. If domain name is provided, resolve to GUID.
+    https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-protocols-oidc#fetch-the-openid-connect-metadata-document
+    """
+    if not _is_guid(namespace.tenant):
+        import requests
+        active_directory_endpoint = cmd.cli_ctx.cloud.endpoints.active_directory
+        url = '{}/{}/.well-known/openid-configuration'.format(active_directory_endpoint, namespace.tenant)
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/profile/_validators.py:30]: `v1.0` endpoint is now deprecated, so we need to migrate to `v2.0`:

https://login.microsoftonline.com/azuresdkteam.onmicrosoft.com/.well-known/openid-configuration ->
https://login.microsoftonline.com/azuresdkteam.onmicrosoft.com/v2.0/.well-known/openid-configuration

Official documentation on `/.well-known/openid-configuration`: https://learn.microsoft.com/en-us/entra/identity-platform/v2-protocols-oidc#find-your-apps-openid-configuration-document-uri

**Pull request merged**